testing/python/approx.py

```diff
@@ -0,0 +1,286 @@
+# encoding: utf-8
+
+import pytest
+import doctest
+
+from pytest import approx
+from operator import eq, ne
+from decimal import Decimal
+from fractions import Fraction
+inf, nan = float('inf'), float('nan')
+
+class MyDocTestRunner(doctest.DocTestRunner):
+
+    def __init__(self):
+        doctest.DocTestRunner.__init__(self)
+
+    def report_failure(self, out, test, example, got):
+        raise AssertionError("'{}' evaluates to '{}', not '{}'".format(
+            example.source.strip(), got.strip(), example.want.strip()))
+
+
+class TestApprox:
```

Comment:
You could use [`@pytest.mark.parametrize`](http://pytest.org/latest/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions) here to separate tests from each other (so e.g. other tests still run if one of them fails), but I'm okay with the current form as well.